lua/entities/gmod_wire_expression2/core/vehicle.lua

```diff
@@ -0,0 +1,146 @@
+--print("Loading Vehicle-Entity support E2 extension...")
+
+if not E2Lib.GetExtensionStatus("vehicle") then
+	hook.Remove("PlayerLeaveVehicle", "E2SetVehicleExitPoint")
+	hook.Remove("EntityRemoved", "E2RemoveVehicleExitPoint")
+end
+
+E2Lib.RegisterExtension("vehicle", true, "todo description.", "todo warning.")
+
+--------------------------------------------------------------------------------
+--  Vehicle-Entity support
+--------------------------------------------------------------------------------
+
+local IsValid = IsValid
+local isOwner = E2Lib.isOwner
+
+-- Localization; Faster access to bit/math/util library functions
+local bit = bit
+local bit_band = bit.band
+local math = math
+local math_max = math.max
+local math_min = math.min
+local util = util
+--local util_TraceHull = util.TraceHull
+local util_PointContents = util.PointContents
+
+registerCallback("e2lib_replace_function", function(funcname, func, oldfunc)
+	if funcname == "IsValid" then
+		IsValid = func
+	elseif funcname == "isOwner" then
+		isOwner = func
+	end
+end)
+
+--------------------------------------------------------------------------------
+
+local wire_pod_exit_distance = GetConVar("wire_pod_exit_distance") -- ConVar is created in Vehicle Exit Point entity. We use this to get the maximum distance an exit-point can move a player (by default it is set to 1000 units).
+local VehicleExitPoints = {} -- This will contain the exit-point data (indexed by Vehicle).
+local InvalidExitValue = { 0, 0, 0 } -- Localized table for "invalid" angle/vector, heck why not, it is better this way than creating a table each time.
```

Comment:
I'd prefer if almost all of the comments in this file were removed - some of them are obviously your unused code, and most of the function descriptions are pretty self-evident from the function signatures themselves.

The only one's I'd suggest keeping are these 3 variable explanations, L106's trace explanation, I guess L122 is ok.